jupyter_book/sphinx.py

```diff
@@ -84,7 +89,7 @@ def build_sphinx(
     # Configuration updates
 
     # Start with the default Sphinx config
-    sphinx_config = DEFAULT_CONFIG.copy()
+    sphinx_config = get_default_config()
```

Comment:
just curious - why is this a better practice than copying the dict?

Reply:
Mutable global variable are BAD! Its one of my grievances with docutils/sphinx, because it makes them not thread-safe and thus incompatible with asyncio

Basically any mutuations to DEFAULT_CONFIG leads to unreliable results during multiple calls, which is what we found is occuring in #896